### PR TITLE
Game status and slice

### PR DESCRIPTION
**Pull request by ocjarman**
This is a pretty big PR.
It touches the flow of the game, updating gamestatus and remaining cards for each team on firebase and redux.

There are some simpler UI changes i.e. 'welcome, username' and 'whos turn it is' on the top of room view

There are also 4 functions critical to the game process: start game, end turn, submit clue, submit answer

on the roomview, whenever there is a change in the 'gameRef', the gamestatus and cards remaining are pulled from firebase to trigger certain actions on the front end

looking forward to going over this and discussing with everyone!

### CODE REVIEW

**Review by rosalie0 — CHANGES_REQUESTED** [review]
plan to refactor

File: src/components/RoomView/Board.jsx [hunk 1]
```diff
@@ -14,9 +26,157 @@ const Board = () => {
...
+    })
+    // get team two operatives data
+    await get(teamTwoOperativesRef).then((snapshot) => {
+      if (snapshot.exists) {
+        teamTwoOps = snapshot.val()
+      }
+    })
+
+  //  if its team 1 ops turn and they are the one who clicked on the card...
+    if (gameStatus === 'team1OpsTurn' && Object.keys(teamOneOps).includes(playerId)) {
```
[the remark this answers is not in the document]
Comment (ocjarman) [reply src/components/RoomView/Board.jsx]: updated! might be cleaner way than pushing ids into an array and checking for playerId but wanted to have something up for team to see later

File: src/components/RoomView/RoomView.jsx [hunk 2]
```diff
@@ -31,12 +33,13 @@ const RoomView = () => {
   // frontend state
   const { playerId, username, roomId, isHost } = useSelector((state) => state.player);
   const { allPlayers } = useSelector((state) => state.allPlayers);
-
+  const [loading, setLoading] = useState(false);
```
Comment (Joshowens16) [inline src/components/RoomView/RoomView.jsx:36]: useState needs to be imported

File: src/components/RoomView/RoomView.jsx [hunk 3]
```diff
@@ -75,29 +84,58 @@ const RoomView = () => {
       }
     });
 
-    // if the player disconnects, remove them from the room
-    onValue(playerNestedInRoomRef, (snapshot) => {
+    onValue(playerNestedInRoom, (snapshot) => {
```
Comment (Joshowens16) [inline src/components/RoomView/RoomView.jsx:87]: should be "playersNestedInRoomRef"

File: src/components/RoomView/setupGame.jsx [hunk 4]
```diff
@@ -9,13 +9,17 @@ function SetupGame() {
   const [selectedWordPackId, setSelectedWordPackId] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
   const roomId = useSelector((state) => state.player.roomId);
-  //   //----------------fet all packs for users to select from-----------------//
+
+  const dispatch = useDispatch();
```
Comment (Joshowens16) [inline src/components/RoomView/setupGame.jsx:13]: useDispatch is not imported, dispatch seems to not be in use in this file so perhaps you could just remove line 13!

File: src/store/index.js [hunk 5]
```diff
@@ -2,14 +2,15 @@ import { configureStore } from '@reduxjs/toolkit';
 import playerReducer from './playerSlice';
```
Comment (Joshowens16) [inline src/store/index.js:2]: redux looks good!

**Review by Joshowens16 — CHANGES_REQUESTED** [review]
Everything looks good, the only things that need be changed are:

1. useState needs to be imported in RoomView
2. on line 87 in RoomView, "playersNestedInRoom" should be "playersNestedInRoomRef"
3. useDispatch in setupGame needs to be imported, although I don't think its being used so you might be able to just remove it. Up to you!

**Review by Joshowens16 — APPROVED** [review]
looks good!

**Review by rosalie0 — APPROVED** [review]

File: src/components/RoomView/Card.jsx [hunk 6]
```diff
@@ -3,7 +3,9 @@ import { useNavigate } from "react-router-dom";
 import { useDispatch, useSelector } from "react-redux";
 import { setWordsInGame } from "../../store/wordsInGameSlice";
 import Grid from "@mui/material/Grid";
-const Card = (singleWord) => {
+
+
+const Card = ({singleWord, value, submitAnswer}) => {
```
[the remark this answers is not in the document]
Comment (rosalie0) [reply src/components/RoomView/Card.jsx]: IGNORE THIS! it was on an old commit and i realized i hadn't finalized my review so it was just sitting there

**Pull request merged**